### ISSUE TITLE
tslint 2.1.0 has regression when using variables-before-functions member ordering rule

### ISSUE THREAD

**Issue by ericvw** [palantir/tslint#292]
Given the following `tslint.json`:

```json
{
    "rules": {
        "member-ordering": [true,
            "variables-before-functions"
        ]
    }
}
```

and `test.ts`:

```typescript
class Foo {
    private bar(): any {
        var bla: { a: string } = {a: '1'};
    }
}
```

`tslint 1.2.0` reports nothing
`tslint 2.1.0` reports: `test.ts[3, 20]: Declaration of public instance member variable not allowed to appear after declaration of private instance member function`

Using `git-bisect` indicates that b26988695f263cda8f518178ae074a7c4d075d9d introduced the regression, but I haven't fully understood why yet.

**Comment by ericvw** [comment]
The key part here that the internal variable is an object literal where `a` is following `bar` that `tslint` is complaining about.

**Comment by ashwinr** [comment]
@gscshoyru can you look into this? thanks.

**Comment by ericvw** [comment]
Thanks!  When may expect a next release that will include this fix?

**Comment by ashwinr** [comment]
we're still fixing a couple of other reported bugs, so tentatively it'll be sometime early next week
